File: src/go/console_util/console_util.go
package console_util

import (
	"bufio"
	"fmt"
	"os"
)

import matrix_calculator "../matrix_calculator"

func Prompt(question string) string {
	fmt.Println(question)
	reader := bufio.NewReader(os.Stdin)
	input, _, _ := reader.ReadLine()
	return string(input)
}

func padLeft(str, pad string, length int) string {
	padding := ""
	for i := 0; i < length-len(str); i++ {
		padding += pad
	}

	return padding + str
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

// DisplayMatrixResult outputs the matrix in the console
func DisplayMatrixResult(result matrix_calculator.MatrixResult) {
	matrix := result.Matrix
	length := max(len(fmt.Sprintf("%d", result.Side*result.Side))+1, 2)

	for rowIndex := 0; rowIndex < result.Side; rowIndex++ {
		rowText := ""
		for columnIndex := 0; columnIndex < result.Side; columnIndex++ {
			rowText += padLeft(fmt.Sprintf("%d", matrix[rowIndex][columnIndex]), " ", length)
		}
		fmt.Println(rowText)
	}

	fmt.Printf("Sum: %v", result.Sum)
}

File: src/fs/consoleUtil.fs
module ConsoleUtil

open System

let Prompt (question:string) : string =
    Console.WriteLine question
    Console.ReadLine()
    
let DisplayMatrixResult (matrix:int[,]) sum side =
    let length = max (String.length (sprintf "%i" (side * side)) + 1) 2
    let indexes = [0..(side - 1)]

    for rowIndex in indexes do
        for columnIndex in indexes do
            printf "%s" ((sprintf "%i" matrix.[rowIndex, columnIndex]).PadLeft(length,' '))
        printfn ""

    printfn "Sum: %i" sum